molo/core/api/importers.py

```diff
@@ -0,0 +1,231 @@
+        )
+    )
+    image.save()
+    return image
+
+
+def separate_fields(fields):
+    """
+    Non-foreign key fields can be mapped to new article instances
+    directly. Foreign key fields require a bit more work.
+    This method returns a tuple, of the same format:
+    (flat fields, nested fields)
+    """
+    flat_fields = {}
+    nested_fields = {}
+
+    # exclude "id" and "meta" elements
+    for k, v in fields.items():
+        if k not in KEYS_TO_EXCLUDE:
+            if type(v) not in [type({}), type([])]:
+                flat_fields.update({k: v})
+            else:
+                nested_fields.update({k: v})
+
+    return flat_fields, nested_fields
+
+
+class PageImporter(object):
+
+    def __init__(self, base_url=None, content=None, content_type=None):
+        self._content_type = content_type
+        self._fields = []
+        self._content = content
+        self._base_url = base_url
+
+    def get_content_from_url(self, base_url):
+        """
+        Sections can have SectionPage and ArticlePage child objects.
+        These have different fields, and thus have to be treated
+        differently.
+        """
+        # assemble url
+        base_url = base_url.rstrip("/")
+        url = base_url + API_PAGES_ENDPOINT + "?type=" + self._content_type + \
+            "&fields=" + ",".join(self._fields) + \
+            "&order=latest_revision_created_at"
+
+        # make request
+        try:
+            response = requests.get(url)
+            self._base_url = base_url
+            self._content = response.json()
+            self._content = self._content["items"]
+            return self._content
+        except requests.exceptions.ConnectionError:
+            return "No content could be found from {}. " \
+                "Are you sure this is the correct URL?".format(base_url)
+        except requests.exceptions.RequestException:
+            return "Content could not be imported at this time. " \
+                   "Please try again later."
+
+    def content(self):
+        if self._content:
+            return self._content
+        return []
+
+    def save(self, indexes, parent_id):
+        pass
+
+
+class ArticlePageImporter(PageImporter):
+
+    def __init__(self, base_url=None, content=None, content_type=None):
+        super(ArticlePageImporter, self).__init__(
+            base_url=base_url, content=content, content_type=content_type
+        )
+        self._content_type = "core.ArticlePage"
+        self._fields = ArticlePage.get_api_fields()
+
+    def save(self, indexes, parent_id):
+        if self.content():
+            parent = Page.objects.get(id=parent_id)
+            for index in indexes:
+                # Remove "id" and "meta" fields
+                selected_article = self.content()[index]
+                fields, nested_fields = separate_fields(selected_article)
+                article = ArticlePage(**fields)
+
+                # TODO: u'related_sections'
```

Comment:
We can make a consideration that `related_sections` can fail silently if they are not being imported.